### PR TITLE
E2E Tests

### PR DESCRIPTION
**Pull request by RaghvindYadav**
## What does this PR do?
This is the new PR is related to e2e testing for signIn, signUp and resetPassword against of #86. 

### Related Issues & PRs
There is no any issues created for it.

#### Key Decisions

- **Implemeneted Approach**:
  I read the svelte official docs and I found for e2e testing playwright would be the best. Here is link: https://svelte.dev/docs/svelte/testing#E2E-tests-using-Playwright
  
- **Alternatives Considered**:
  We could also use other solutions like [Cypress](https://www.cypress.io/) or [NightwatchJS](https://nightwatchjs.org/).
- **Rationale**:
  I decide to use playwright because it was easy to create and our app was already configured for playwright.

## Breaking Changes

- [ ] This PR includes breaking changes
  

### Limitations & Concerns

- **Known limitations**:
- Make sure no any account exist in the db with this email id e2e@test.com. 
- **Edge cases**:
- **Potential risks**:

### How to Test

1. Prerequisites:
  You need to install [playwright vscode extension](https://playwright.dev/docs/getting-started-vscode). 
2. Test steps:
   Go to the testing tab in vs code and find the playwright option. You can run the test manual by using the below terminal code for the specific test. 
   `npx playwright test tests/e2e/signin.spec.ts --ui `
   
   Just rename the file name for other tests.
3. Expected results:
   Every E2E test should pass.

## Final Checklist
- [x] Linked related issues
- [x] Rebased with target branch
- [ ] No lint errors
- [x] QA'd in multiple browsers
- [ ] Added/updated tests (if applicable)
- [ ] Mobile responsive (if applicable)
- [x] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage

- [ ] AI tools were used in development
  - **Tool used**:
    Augment code
  - **Prompts used**:
    
    - **Initial prompt**: create a e2e test for signIn and signup, please write the test case carefully here are the full code of signIn and signUp @sign-up-form.svelte and @sign-in-form.svelte.
- [ ] No AI tools were used
  - **Reason**:
    To make some test get passed I need to add some component Id in their respective file.

### CODE REVIEW

File: tests/e2e/signup.spec.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  // Click resend button
+  await expect(resendButton).toBeEnabled();
+  await resendButton.click();
+
+  // Verify the resend timer is reset
+  await expect(resendButton).toBeDisabled();
+
+  // Clean up: Delete the test account
+  await deleteTestAccount(page);
+});
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply tests/e2e/signup.spec.ts:124]: It already specific.

File: src/lib/contexts/my-user-context.svelte.ts [hunk 2]
```diff
@@ -565,3 +565,7 @@ export class MyUserContext {
 
 // Create a singleton instance
 export const myUserContext = new MyUserContext();
+
+if (typeof window !== 'undefined') {
+  (window as Window & typeof globalThis & { __myUserContext: typeof myUserContext }).__myUserContext = myUserContext;
+}
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply src/lib/contexts/my-user-context.svelte.ts:571]: I agreed with your concern but my test window doesn't accept direct call of context functions.

File: tests/e2e/signup.spec.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  // Click sign up button
+  await page.locator('#form-button').click();
+
+  // Wait for verification step to appear
+  const otpInputs = page.locator('#verification-code');
+  await expect(otpInputs.first()).toBeVisible();
+
+  await page.waitForTimeout(30000);
+  // Wait for the resend timer to expire (in a real test, you might want to mock this)
+  // For testing purposes, we'll just check if the resend button becomes enabled
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply tests/e2e/signup.spec.ts:110]: I tried to skip the waiting but it not works.

```
await page.evaluate(() => {
  window.setTimeout = ((original) => (fn: TimerHandler, ms: number, ...args: any[]) => {
    if (ms >= 30000) return original(fn, 0, ...args);
    return original(fn, ms, ...args);
  })(window.setTimeout) as typeof window.setTimeout;

  window.setInterval = ((original) => (fn: TimerHandler, ms: number, ...args: any[]) => {
    if (ms >= 30000) return original(fn, 0, ...args);
    return original(fn, ms, ...args);
  })(window.setInterval) as typeof window.setInterval;
});
```

File: tests/e2e/utils/delete-test-account.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          return false;
+        }
+      } catch (error) {
+        console.error('Failed to delete test account:', error);
+        return false;
+      }
+    },
+    { reason, description },
+  );
+}
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply tests/e2e/utils/delete-test-account.ts:45]: direct call throw errors

File: tests/e2e/signin.spec.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+  // Click verify button
+  await page.getByRole('button', { name: 'Verify' }).click();
+
+  // Verify redirection to home page
+  await expect(page).toHaveURL('/');
+
+  // Clean up: Delete the test account
+  await deleteTestAccount(page);
+});
```
[the remark this answers is not in the document]
Comment (RaghvindYadav) [reply tests/e2e/signin.spec.ts:143]: Resend button will active after 30 seconds.

File: src/lib/contexts/my-user-context.svelte.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonahtwalker) [reply src/lib/contexts/my-user-context.svelte.ts:571]: I'm not sure I'm following here, why do we need to include this? I have been expecting to see all of the E2E tests that touch the client fail because the client cannot be initialized, or does not have a connection.

**Review by jonahtwalker — CHANGES_REQUESTED** [review]
Thanks Raghav, these are looking good and clean.

Why do we need the additional window logic for the MyUserContext?

I don't have other changes right now, but think we should get rid of that. The tests should likely fail for a client error, not that we can't find the OTP input
